warehouse/query-core/src/main/java/datawave/query/iterator/logic/ContentSummaryIterator.java

```diff
@@ -0,0 +1,329 @@
+                }
+            } else {
+                // we had an end document specified in the end key, so end there
+                endKey = new Key(range.getStartKey().getRow(), Constants.D_COLUMN_FAMILY, new Text(dtAndUid));
+            }
+        }
+        log.debug("{} seek'ing to end key: {}", this, endKey);
+
+        // if we have actually exhausted our range, then return with no next key
+        if (endKey != null && startKey != null && endKey.compareTo(startKey) <= 0) {
+            this.scanRange = null;
+            this.tk = null;
+            this.tv = null;
+            return;
+        }
+
+        // set our d keys scan range
+        this.scanRange = new Range(startKey, false, endKey, false);
+
+        if (log.isDebugEnabled()) {
+            log.debug("{} seek'ing to: {} from requested range {}", this, this.scanRange, range);
+        }
+
+        // seek the underlying source
+        source.seek(this.scanRange, D_COLUMN_FAMILY_BYTE_SEQUENCE, true);
+
+        // get the next key
+        next();
+    }
+
+    @Override
+    public void next() throws IOException {
+        tk = null;
+        tv = null;
+
+        if (log.isTraceEnabled()) {
+            log.trace("{} calling next on {}", source.hasTop(), scanRange);
+        }
+
+        // find a valid dt/uid (depends on initial column families set in seek call)
+        String dtUid = null;
+        while (source.hasTop() && dtUid == null) {
+            Key top = source.getTopKey();
+            String thisDtUid = getDtUidFromDKey(top);
+            // if this dt and uid are in the accepted column families...
+            if (columnFamilies.contains(thisDtUid) == inclusive) {
+                // we can use this document
+                dtUid = thisDtUid;
+            } else {
+                seekToNextUid(top.getRow(), thisDtUid);
+            }
+        }
+
+        // if no more d keys, then we are done.
+        if (!source.hasTop() || dtUid == null) {
+            return;
+        }
+
+        Key top = source.getTopKey();
+
+        // this is where we will save all the content found for this document
+        final Map<String,byte[]> foundContent = new HashMap<>();
+
+        // while we have d keys for the same document
+        while (source.hasTop() && dtUid.equals(getDtUidFromDKey(source.getTopKey()))) {
+            top = source.getTopKey();
+
+            // get the view name
+            String currentViewName = getViewName(top);
+
+            for (String name : viewSummaryOrder) {
+                if (name.endsWith("*")) {
+                    name = name.substring(0, name.length() - 1);
+                    if (currentViewName.startsWith(name)) {
+                        foundContent.put(currentViewName, source.getTopValue().get());
+                    }
+                } else {
+                    if (currentViewName.equalsIgnoreCase(name)) {
+                        foundContent.put(currentViewName, source.getTopValue().get());
```

Comment:
should there be a limit on the amount of content that will be added to prevent potential memory problems? This might also require some more configuration flags

Reply:
current implementation is if the content is over X size, it will immediately decode/decompress and truncate and save that smaller value vs saving the whole content and dealing with it later